### ISSUE TITLE
Snowflake: create procedures opening bracket unparsable section error

### ISSUE THREAD

**Issue by myschkyna** [sqlfluff/sqlfluff#1622]
## Expected Behaviour
I expected a create procedure statement in the Snowflake dialect to work

## Observed Behaviour
Opening brackets in procedure statement generate unparsable section error, see lint output below:
```bash
== [test_procedure.sql] FAIL
L:   1 | P:  50 |  PRS | Line 1, Position 50: Found unparsable section: '
                       | (test_table VARCHAR(), test_col VARCHAR...'
```

Parser output:
```bash
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            binary_operator:                                  'OR'
[L:  1, P: 10]      |            whitespace:                                       ' '
[L:  1, P: 11]      |            keyword:                                          'REPLACE'
[L:  1, P: 18]      |            whitespace:                                       ' '
[L:  1, P: 19]      |            keyword:                                          'PROCEDURE'
[L:  1, P: 28]      |            whitespace:                                       ' '
[L:  1, P: 29]      |            object_reference:
[L:  1, P: 29]      |                identifier:                                   'public'
[L:  1, P: 35]      |                dot:                                          '.'
[L:  1, P: 36]      |                identifier:                                   'test_procedure'
[L:  1, P: 50]      |        unparsable:                                           !! Expected: 'Nothing...'
[L:  1, P: 50]      |            whitespace:                                       ' '
[L:  1, P: 51]      |            bracketed:
[L:  1, P: 51]      |                start_bracket:                                '('
[L:  1, P: 52]      |                [META] indent:
[L:  1, P: 52]      |                expression:
[L:  1, P: 52]      |                    column_reference:
[L:  1, P: 52]      |                        identifier:                           'test_table'
[L:  1, P: 62]      |                whitespace:                                   ' '
[L:  1, P: 63]      |                expression:
[L:  1, P: 63]      |                    function:
[L:  1, P: 63]      |                        function_name:
[L:  1, P: 63]      |                            function_name_identifier:         'VARCHAR'
[L:  1, P: 70]      |                        bracketed:
[L:  1, P: 70]      |                            start_bracket:                    '('
[L:  1, P: 71]      |                            end_bracket:                      ')'
[L:  1, P: 72]      |                comma:                                        ','
[L:  1, P: 73]      |                whitespace:                                   ' '
[L:  1, P: 74]      |                expression:
[L:  1, P: 74]      |                    column_reference:
[L:  1, P: 74]      |                        identifier:                           'test_col'
[L:  1, P: 82]      |                whitespace:                                   ' '
[L:  1, P: 83]      |                expression:
[L:  1, P: 83]      |                    function:
[L:  1, P: 83]      |                        function_name:
[L:  1, P: 83]      |                            function_name_identifier:         'VARCHAR'
[L:  1, P: 90]      |                        bracketed:
[L:  1, P: 90]      |                            start_bracket:                    '('
[L:  1, P: 91]      |                            end_bracket:                      ')'
[L:  1, P: 92]      |                [META] dedent:
[L:  1, P: 92]      |                end_bracket:                                  ')'
[L:  1, P: 93]      |            newline:                                          '\n'
[L:  2, P:  1]      |            raw:                                              'RETURNS'
[L:  2, P:  8]      |            whitespace:                                       ' '
[L:  2, P:  9]      |            raw:                                              'VARCHAR'
[L:  2, P: 16]      |            bracketed:
[L:  2, P: 16]      |                start_bracket:                                '('
[L:  2, P: 17]      |                end_bracket:                                  ')'
[L:  2, P: 18]      |            newline:                                          '\n'
[L:  3, P:  1]      |            raw:                                              'LANGUAGE'
[L:  3, P:  9]      |            whitespace:                                       ' '
[L:  3, P: 10]      |            raw:                                              'JAVASCRIPT'
[L:  3, P: 20]      |            newline:                                          '\n'
[L:  4, P:  1]      |            raw:                                              'AS'
[L:  4, P:  3]      |            newline:                                          '\n'
[L:  5, P:  1]      |            raw:                                              '$$  \ntry {\n    var sql_command = "ALTER TABLE " + test_table + " DROP " + tet_col;\n    snowflake.execute ({sqlText: sql_command});\n    return "Succeeded.";\n}\ncatch (err) {\n   return "Failed: execute "+ sql_command +". Error : "+ err;   // Return a success/error indicator.\n}\n$$'
[L: 14, P:  3]      |    newline:                                                  '\n'
[L: 15, P:  1]      |    statement_terminator:                                     ';'
[L: 15, P:  2]      |    newline:                                                  '\n'
```

## Steps to Reproduce
1. Install sqlfluff package with version version 0.7.0a4
2. Create file with contents below
3. Run sqlfluff lint <path-to-file>

Content of SQL file test_procedure.sql:
```sql
CREATE OR REPLACE PROCEDURE public.test_procedure (test_table VARCHAR(), test_col VARCHAR())
RETURNS VARCHAR()
LANGUAGE JAVASCRIPT
AS
$$  
try {
    var sql_command = "ALTER TABLE " + test_table + " DROP " + tet_col;
    snowflake.execute ({sqlText: sql_command});
    return "Succeeded.";
}
catch (err) {
   return "Failed: execute "+ sql_command +". Error : "+ err;   // Return a success/error indicator.
}
$$
;

```

## Dialect
Snowflake

## Version
sqlfluff, version 0.7.0a4

## Configuration
```
[sqlfluff]
dialect = snowflake
# These rules are disable to allow: 
# dbt templating does not keep trailing new lines (L009)
# L031: use of table alias in join clause
# L011: implicit alias of table
# L034: any order of select statements
# L029: keywords as identifiers
exclude_rules = L009,L011,L029,L031,L034

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = upper

[sqlfluff:rules:L014]  # Tables, columns, views
extended_capitalisation_policy = lower

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = upper

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = upper

[sqlfluff:rules:L016] # Max line length
max_line_length = 120
```

**Comment by WittierDinosaur** [comment]
To add some context here - the entire CREATE PROCEDURE block appears to be missing

**Comment by myschkyna** [comment]
@WittierDinosaur I think the create procedure can be found in the CreateStatementSegment class in src/sqlfluff/dialects/snowflake_dialect.py. It is wrapped in a OneOf clause.
